### ISSUE TITLE
FailedPreconditionError

### ISSUE THREAD

**Issue by Firuz-JuraevML** [Trusted-AI/adversarial-robustness-toolbox#1159]
Hello, 
I am using ART for testing the robustness of the models. Thank you for that library, it is amazing! 

However, I have some errors that I could not solve. When I use the pre-trained models from Keras (VGG16, MobilNet). While applying attacks like CW, it is giving the following error: 

**FailedPreconditionError**: Could not find variable block1_conv1/kernel. This could mean that the variable has been deleted. In TF1, it can also mean the variable is uninitialized. Debug info: container=localhost, status=Not found: Container localhost does not exist. (Could not find resource: localhost/block1_conv1/kernel)
	 [[{{node block1_conv1/Conv2D/ReadVariableOp}}]]

I really appreciate your support! 
Thank you.

**Comment by beat-buesser** [comment]
Hi @Firuz-JuraevML Thank you very much for using ART and you compliments, we appreciate them very much!

Would you have a short code example that reproduces the error which you could post here or in a Google Colab notebook ?

**Comment by Firuz-JuraevML** [comment]
Hello @beat-buesser, Thank you so much for your response. 

Here is my code, you can enter and try it. 
Colab Link: https://colab.research.google.com/drive/1vBPhrs_NTOH1yKSp1trp8U6vyET2NJBK?usp=sharing

**Comment by beat-buesser** [comment]
Hi @Firuz-JuraevML Thank you for the notebook, this is very helpful. I think this might be a bug in `KerasClassifier.class_gradients` that we need to investigate, I was able to reproduce it outside of the notebook. Therefore, I have changed the notebook to use `TensorFlowV2Classifier` instead and it seems to work. Other changes include applying `preprocessing_defences`  for the preprocessor and changing VGG16 to predict logits fas expected by the C&W attack.
